package/MDAnalysis/coordinates/memory.py

```diff
@@ -310,6 +309,16 @@ def parse_n_atoms(filename, order='fac', **kwargs):
         # assume filename is a numpy array
         return filename.shape[order.find('a')]
 
+    def copy(self):
+        """Return a copy of this Memory Reader"""
+        return self.__class__(
+            self.coordinate_array.copy(),
+            order=self.stored_order,
+            dimensions=self.ts.dimensions,
+            dt=self.ts.dt,
+            filename=self.filename,
+        )
```

Comment:
Here is an inconsistency with the file based readers. The new memory reader will start at frame 0 and have a timestep object of frame 0. This should also be changed to seek to the current frame of the source reader.